### DISCUSSION TITLE
zh/guide/feature/search

### DISCUSSION THREAD

**Discussion by giscus** [vuepress-theme-hope/giscus-discussions#468] [Announcements]
# zh/guide/feature/search

主题对 、 [@vuepress/plugin-docsearch][docsearch] 和 [@vuepress/plugin-search][search] 提供了内置支持。你只需要添加并配置所需的搜索插件，就能够在导航栏获得一个搜索框。 如果你需要搜索插件，请通过 VuePress 配置文件 的 plugins 选项应用它。 注意 主题只是添加了上述搜索插件的支持，而并没有捆绑它们，你需要自己安装和调用。 警告 请勿在主题选项中使用 plugins.search。 由于主题只能调用它捆绑的插件，因此主题选项中的 plugins 字段仅接受特定插件名称。

https://theme-hope.vuejs.press/zh/guide/feature/search.html

**Comment by Huxzhi** [comment]
你好， 您 vuepress-theme-hope模板内置 的插件加载方式和 该案例不同，不是 plugins: [ ] ，请问怎么加载插件
```js
 plugins: {
    blog: true,

    comment: {
      // @ts-expect-error: You should generate and use your own comment service
      provider: "Giscus",
    },
   ...
```

**Reply by Mister-Hope** [reply]
-  ![image](https://user-images.githubusercontent.com/33315834/220048136-36c03739-9e75-4203-a947-68f9b00792c8.png)

- https://theme-hope.vuejs.press/zh/cookbook/tutorial/config.html#%E9%85%8D%E7%BD%AE%E6%96%87%E4%BB%B6

- https://theme-hope.vuejs.press/zh/config/intro.html

## So tell me, what else can I do to prevent noob like you posting question like this?🫤

**Comment by Huxzhi** [comment]
哦哦，我搞错了，我在 theme.ts 内配置选项。应该在  config.ts 文件里配置
